socorro/processor/general_transform_rules.py

```diff
@@ -7,6 +7,45 @@
 from socorro.processor.rules.base import Rule
 
 
+class DeNullRule(Rule):
+    """Removes nulls from keys and values
+
+    Sometimes crash reports come in with junk data. This removes the egregious
+    junk that causes downstream processing and storage problems.
+
+    NOTE(willkg): While this changes the raw crash, that raw crash should never
+    get saved back to storage, so this doesn't overwrite the original material.
+
+    """
+    def de_null(self, s):
+        """Remove nulls from bytes and str values
+
+        :arg str/bytes value: The str or bytes to remove nulls from
+
+        :returns: str or bytes without nulls
+
+        """
+        if isinstance(s, bytes) and b'\0' in s:
+            return s.replace(b'\0', b'')
+
+        if isinstance(s, str) and '\0' in s:
+            return s.replace('\0', '')
+
+        # If it's not a bytes or a str, it's something else and we should
+        # return it as is
+        return s
```

Comment:
This is identical to the code in antenna now. I spent some time looking at whether checking for null bytes and then replacing was faster than just replacing.

Just replacing:

```
$ python3 -m timeit -n 10000 -s "s = 'a' * 100000" "s.replace('\0', '')"
10000 loops, best of 3: 26.1 usec per loop
```

Test and then replace:

```
$ python3 -m timeit -n 10000 -s "s = 'a' * 100000" "if '\0' in s: s.replace('\0', '')"
10000 loops, best of 3: 0.986 usec per loop
```

I think the data we're fixing generally doesn't have null characters--it's all key/val string data from the `.extra` and annotations in the crash report. So test-and-replace it is!